src/main/java/com/useful/ucars/uCarsListener.java

```diff
@@ -562,13 +568,15 @@ public void tickCalcsAndLegacy(VehicleUpdateEvent event) {
 				UEntityMeta.removeMetadata(vehicle, "car.vec"); //Update the 'car.vec' metadata with an otherwise identical event; but without the player object attached
 				ucarUpdateEvent et = new ucarUpdateEvent(vehicle, evt.getTravelVector().clone(), null, evt.getDir()); //Clone of the other event, except no player object attached
 				et.setRead(evt.getReadCount()); //Make sure it IS a clone (With correct variable values)
+				if(vehicle.hasMetadata("car.frozen")) {
```

Comment:
Why is this required? The uCarUpdateEvent I think should still be fired for a frozen car

Reply:
This is currently needed as cars that should be frozen can drive when you leave this out.

I'm gonna look into placing this check somewhere else

Reply:
okay, in the update event would make more sense I think. There should have already been a check somewhere?

Reply:
There is one in calculateCarStats which is called during the update event.
The thing is that (at least with race controls) this does not work.
I'm gonna look into it